dulwich/archive.py

```diff
@@ -69,7 +83,7 @@ def read(self, maxbytes=None):
         return b"".join(buf)
 
 
-def tar_stream(store, tree, mtime, prefix=b"", format=""):
+def tar_stream(store, tree, mtime, prefix=b"", compression_format=""):
```

Comment:
I will modify the parameters of other modules to call this function to modify. I changed the parameter name because it covered the `format` function

Reply:
External callers may use it as well - just changing the callers in the dulwich codebase is not sufficient.

Reply:
I have a call to modify the test function.

Reply:
That's fine, it's just a warning - we don't use format in this function.

Reply:
So can I uniformly modify `format` to become `compression_format`?

Reply:
No, please don't change this - it's not fixing any actual problems.

Reply:
But I don't like this parameter name very much. I think this modification can be used as a modification of the updated version of `x` in `vx.y.z`, so that it is acceptable even if the API is modified.

Reply:
I don't see what's wrong with the current parameter name, and I don't think changing it is worth the disruption to API users.

compression_format is an incorrect name FWIW, since this is not just about compression but about the container format as well.

Reply:
if change nane `format` to `container_format`?

Reply:
If it's not good, I will change it back

Reply:
Why does it need to be renamed? It just causes disruption for API users. It shadows a builtin and we should be careful about that, but otherwise it's fine.

Reply:
I thought of one. If I keep two at the same time, the parameter can be kept for a period of time, but at the same time, a time reminder is provided to cancel the `format` because it covers the original built-in function.

Reply:
That still requires callers change. The current parameter name is fine, I don't see why we need to change it.

Reply:
I changed the parameter name because it covered the format function. I have changed it back, but I still suggest not to overwrite the original built-in functions.